#! /bin/sh /usr/share/dpatch/dpatch-run
## 10-fix-proj-install.dpatch by Marek Brudka <mbrudka@aster.pl>
##
## DP: prj_install.pl simplifies installation 

@DPATCH@
--- ACE_wrappers.orig/MPC/prj_install.pl	2008-08-20 10:38:13.000000000 +0200
+++ ACE_wrappers/MPC/prj_install.pl	2010-03-25 15:43:52.000000000 +0100
@@ -1,3 +1,4 @@
+#!/usr/bin/perl
 #! /usr/bin/perl
 eval '(exit $?0)' && eval 'exec perl -w -S $0 ${1+"$@"}'
     & eval 'exec perl -w -S $0 $argv:q'
@@ -123,7 +124,7 @@
     $insdir = '';
   }
 
-  my $odir = ($dir eq '' ? '.' : $dir) . '/' . $insdir;
+  my $odir = ($dir eq '' ? './' . $insdir: $dir) ;
   if ($tag eq 'exe_output') {
     my @exes;
     my $fh   = new FileHandle();
@@ -131,7 +132,7 @@
       foreach my $file (grep(!/^\.\.?$/, readdir($fh))) {
         if ($file =~ /^$name$/ ||
             $file =~ /^$name.*\.exe$/i) {
-          push(@exes, "$dir$insdir$file");
+          push(@exes, "$dir$file");
         }
       }
       closedir($fh);
@@ -145,7 +146,7 @@
       foreach my $file (grep(!/^\.\.?$/, readdir($fh))) {
         if ($file =~ /^lib$name\.(a|so|sl)/ ||
             $file =~ /^(lib)?$name.*\.(dll|lib)$/i) {
-          push(@libs, "$dir$insdir$file");
+          push(@libs, "$dir$file");
         }
       }
       closedir($fh);
